test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests.cs

```diff
@@ -915,6 +971,36 @@ private async Task<HttpResponseMessage> GetHostStatusAsync()
             return await _fixture.Host.HttpClient.SendAsync(request);
         }
 
+        private async Task SetHostStateAsync(string state)
+        {
+            var masterKey = await _fixture.Host.GetMasterKeyAsync();
+            var request = new HttpRequestMessage(HttpMethod.Put, "admin/host/state");
+            request.Content = new StringContent(state);
+            request.Content.Headers.ContentType = new MediaTypeHeaderValue("text/plain");
+            request.Headers.Add(AuthenticationLevelHandler.FunctionsKeyHeaderName, masterKey);
+            var response = await _fixture.Host.HttpClient.SendAsync(request);
+            Assert.Equal(response.StatusCode, HttpStatusCode.Accepted);
+        }
+
+        private async Task AwaitHostStateAsync(ScriptHostState state)
+        {
+            await TestHelpers.Await(async () =>
+            {
+                var response = await GetHostStatusAsync();
+                var hostStatus = response.Content.ReadAsAsync<HostStatus>();
+                return string.Compare(hostStatus.Result.State, state.ToString(), StringComparison.OrdinalIgnoreCase) == 0;
+            });
+        }
+
+        private async Task<HttpResponseMessage> RestartHostAsync()
+        {
```

Comment:
Any reason not to get the `IScriptHostManager` from the services and call the restart directly here?

Reply:
No reason really, other than all these other helpers are operating through admin apis. Also we have the /restart API with no tests currently I believe